### PR TITLE
jenkins: preparations to upload and serve artifacts

### PR DESCRIPTION
**Pull request by flokli**
Depends on #160, will be rebased once merged (so ignore the first 2 commits).

This adds the necessary services running on the jenkins-controller machine to upload and serve artifacts from Azure storage.

 - A per-environment storage container for artifacts is added via terraform
 - Two rclone services exposing the bucket contents, rw and ro, plus config to expose the ro part via caddy.
 - Our rclone gains another patch/feature, support to connect to webdav backends via unix domain sockets, and that rclone is made available in jenkins' `$PATH`.

### CODE REVIEW

**Comment by henrirosten** [comment]
@flokli: On testing this change on a new deployment, the Jenkins service seems to boot-up to the setup wizard. Is that expected?

**Comment by flokli** [comment]
Hmm, I only tested with my installation, keeping its state. We might need to set `-Djenkins.install.runSetupWizard=false` as per [here](https://github.com/jenkinsci/configuration-as-code-plugin/issues/393#issuecomment-408122043).

**Comment by flokli** [comment]
Disabling the setup wizard is done in https://github.com/tiiuae/ghaf-infra/pull/160, this is unrelated to this PR.

**Comment by flokli** [comment]
Drafting until #160 is merged.

**Comment by flokli** [comment]
With https://github.com/tiiuae/ghaf-infra/pull/160 merged, I rebased this.

**Comment by flokli** [comment]
Corresponding pipeline PR: https://github.com/tiiuae/ghaf-jenkins-pipeline/pull/25

**Comment by henrirosten** [comment]
Reviewed and verified working by deploying to a private environment.

I guess once this is merged, and all the Jenkins pipelines are changed to no longer use the `archiveArtifacts`, we could also downsize the [Jenkins disk](https://github.com/tiiuae/ghaf-infra/blob/e0fe744217d49ba643207b7fe982e872907157ba/terraform/jenkins-controller.tf#L136) to something more reasnoable. But that needs to happen in a follow-up PR later.

**Review by henrirosten — APPROVED** [review]

**Comment by flokli** [comment]
Yeah, that's blocked on everything being switched over, and we might need to schedule some time to properly sync some existing state (if we want to keep the disk contents)

**Pull request merged**
